src/Compilers/CSharp/Portable/Binder/InContainerBinder.cs

```diff
@@ -117,40 +105,21 @@ internal override TypeWithAnnotations GetIteratorElementType()
         {
             Debug.Assert(result.IsClear);
 
-            if (IsSubmissionClass)
-            {
-                this.LookupMembersInternal(result, _container, name, arity, basesBeingResolved, options, originalBinder, diagnose, ref useSiteInfo);
-                return;
-            }
-
             // first lookup members of the namespace
             if ((options & LookupOptions.NamespaceAliasesOnly) == 0)
             {
                 this.LookupMembersInternal(result, _container, name, arity, basesBeingResolved, options, originalBinder, diagnose, ref useSiteInfo);
 
                 if (result.IsMultiViable)
                 {
-                    if (arity == 0 && _container.IsNamespace)
+                    if (arity == 0)
                     {
                         // symbols cannot conflict with using alias names
-                        Binder binder = this;
-
-                        do
-                        {
-                            binder = binder.Next;
-                        }
-                        while (binder is object && !(binder is ImportsBinder));
-
-                        if (binder is ImportsBinder importsBinder && importsBinder.Container == _container)
+                        if (Next is WithExternAndUsingAliasesBinder withUsingAliases && withUsingAliases.IsUsingAlias(name, originalBinder.IsSemanticModelBinder, basesBeingResolved))
```

Comment:
>Next is WithExternAndUsingAliasesBinder [](start = 28, length = 39)

Do we have a guarantee somewhere that a `WithExternalAndUsingAliasesBinder` is truly the next binder, and not somewhere further along the chain? I'm a little concerned that a future refactoring could break that invariant, so it would be good to ensure we have a debug assert covering it.

Reply:
> Do we have a guarantee somewhere that a WithExternalAndUsingAliasesBinder is truly the next binder, and not somewhere further along the chain? I'm a little concerned that a future refactoring could break that invariant, so it would be good to ensure we have a debug assert covering it.

There is no such guarantee. It is there when appropriate and the Binder factory is responsible to make this happen.

---
In reply to: [588673669](https://github.com/dotnet/roslyn/pull/51676#discussion_r588673669) [](ancestors = 588673669)